internal/core/operations/healthcontroller/health_controller_executor_test.go

```diff
@@ -0,0 +1,518 @@
+// MIT License
+//
+// Copyright (c) 2021 TFG Co
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package healthcontroller_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/topfreegames/maestro/internal/core/services/room_manager"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	ismock "github.com/topfreegames/maestro/internal/adapters/instance_storage/mock"
+	"github.com/topfreegames/maestro/internal/core/entities"
+	"github.com/topfreegames/maestro/internal/core/entities/forwarder"
+	"github.com/topfreegames/maestro/internal/core/entities/game_room"
+	"github.com/topfreegames/maestro/internal/core/entities/operation"
+	"github.com/topfreegames/maestro/internal/core/operations/add_rooms"
+	"github.com/topfreegames/maestro/internal/core/operations/healthcontroller"
+	"github.com/topfreegames/maestro/internal/core/operations/remove_rooms"
+	mockports "github.com/topfreegames/maestro/internal/core/ports/mock"
+)
+
+func TestSchedulerHealthController_Execute(t *testing.T) {
+	type ExecutionPlan struct {
+		PlanMocks func(
+			roomStorage *mockports.MockRoomStorage,
+			instanceStorage *ismock.MockGameRoomInstanceStorage,
+			schedulerStorage *mockports.MockSchedulerStorage,
+			operationManager *mockports.MockOperationManager,
+		)
+		ShouldFail bool
```

Comment:
```suggestion
		planMocks func(
			roomStorage *mockports.MockRoomStorage,
			instanceStorage *ismock.MockGameRoomInstanceStorage,
			schedulerStorage *mockports.MockSchedulerStorage,
			operationManager *mockports.MockOperationManager,
		)
		shouldFail bool
```

Reply:
@caiordjesus Any reason for using exported fields here and below?

Reply:
Not really, very well observed